From 0a578e9cbdaf8c7de49f82048dc63ac65958f018 Mon Sep 17 00:00:00 2001
From: Piioo <Piotr.Szal@googlemail.com>
Date: Tue, 24 Jun 2014 15:21:04 +0200
Subject: [PATCH] redis add top_topics_count per period

---
 app/models/site_setting.rb | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/app/models/site_setting.rb b/app/models/site_setting.rb
index 98dfe84..cd2e78f 100644
--- a/app/models/site_setting.rb
+++ b/app/models/site_setting.rb
@@ -88,10 +88,16 @@ class SiteSetting < ActiveRecord::Base
 
   def self.has_enough_topics_to_redirect_to_top
     TopTopic.periods.each do |period|
-      topics_per_period = TopTopic.where("#{period}_score > 0")
-                                  .limit(SiteSetting.topics_per_period_in_top_page)
-                                  .count
-      return true if topics_per_period >= SiteSetting.topics_per_period_in_top_page
+
+      topics_per_period = $redis.get("top_topics_per_#{period}_count")
+
+      if topics_per_period.blank?
+        topics_per_period = TopTopic.where("#{period}_score > 0")
+                                    .limit(SiteSetting.topics_per_period_in_top_page)
+                                    .count
+        $redis.setex("top_topics_per_#{period}_count", 5.minutes, topics_per_period)
+      end
+      return true if topics_per_period.to_i >= SiteSetting.topics_per_period_in_top_page
     end
     # nothing
     false
-- 
1.8.5.2 (Apple Git-48)

