From 5be3395a7c8e1273c057b8e4c14aba6f14979ba2 Mon Sep 17 00:00:00 2001
From: Benjamin Kampmann <ben.kampmann@gmail.com>
Date: Mon, 31 Mar 2014 12:31:57 +0200
Subject: [PATCH 5/8] typos and symbols instead of strings

---
 lib/archetype.rb | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/lib/archetype.rb b/lib/archetype.rb
index 10fc266..a104f00 100644
--- a/lib/archetype.rb
+++ b/lib/archetype.rb
@@ -54,17 +54,18 @@ class Archetype
 
   def self.register(name, options={})
     @archetypes ||= {}
-    archetype = options["archetype"] || Archetype.new(name, options)
+    archetype = options[:archetype] || Archetype.new(name, options)
     @archetypes[name] = archetype
-    if archetype.options.has_key? :capabilties
-      archetype.options[:capabilties].each do |cap|
+    if archetype.options.has_key? :capabilities
+      archetype.options[:capabilities].each do |cap|
+        puts cap, name
         register_capability(cap, name)
       end
     end
   end
 
   # By default we have a regular archetype
-  register 'regular', capabilties: [:searchable, :shown_publicly, :linkable]
+  register 'regular', capabilities: [:searchable, :shown_publicly, :linkable]
 
 end
 
-- 
1.8.3.4 (Apple Git-47)

