From 55ce6d7a1f4141e0cd1c12334abbd9710ddab3bf Mon Sep 17 00:00:00 2001
From: Benjamin Kampmann <ben.kampmann@gmail.com>
Date: Mon, 31 Mar 2014 16:56:06 +0200
Subject: [PATCH 07/18] make archetype calls from the topic model

---
 app/models/topic.rb | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/app/models/topic.rb b/app/models/topic.rb
index e620d3f..af0e561 100644
--- a/app/models/topic.rb
+++ b/app/models/topic.rb
@@ -28,6 +28,11 @@ class Topic < ActiveRecord::Base
     2**31 - 1
   end
 
+  def archetype_handle(func, *args, **kwargs)
+    @archetypeObj ||= Archetype.get_archetype(self.archetype)
+    @archetypeObj.__send__ func, *args, **kwargs
+  end
+
   def featured_users
     @featured_users ||= TopicFeaturedUsers.new(self)
   end
-- 
1.8.3.4 (Apple Git-47)

