From 0d1a29351f820836a753b0c0de4516e95db9eacf Mon Sep 17 00:00:00 2001
From: Benjamin Kampmann <ben.kampmann@gmail.com>
Date: Mon, 19 May 2014 11:45:56 +0200
Subject: [PATCH 14/18] archetype defaults on initial load

---
 config/routes.rb | 27 ++++++++++++++-------------
 1 file changed, 14 insertions(+), 13 deletions(-)

diff --git a/config/routes.rb b/config/routes.rb
index 0fb0f47..d1268db 100644
--- a/config/routes.rb
+++ b/config/routes.rb
@@ -264,7 +264,7 @@ Discourse::Application.routes.draw do
   # We've renamed popular to latest. If people access it we want a permanent redirect.
   get "popular" => "list#popular_redirect"
 
-  def categorizer(suffix)
+  def categorizer(archetype)
     resources :categories, :except => :show
     get "category/:id/show" => "categories#show"
     post "category/:category_id/move" => "categories#move"
@@ -274,27 +274,27 @@ Discourse::Application.routes.draw do
     get "category/:category/none" => "list#category_none_latest"
     get "category/:parent_category/:category" => "list#parent_category_category_latest"
 
-    get "top" => "list#top"
-    get "category/:category/l/top" => "list#category_top", as: "category_top#{suffix}"
-    get "category/:category/none/l/top" => "list#category_none_top", as: "category_none_top#{suffix}"
-    get "category/:parent_category/:category/l/top" => "list#parent_category_category_top", as: "parent_category_category_top#{suffix}"
+    get "top" => "list#top", defaults: { archetype: archetype }
+    get "category/:category/l/top" => "list#category_top", as: "category_top#{archetype}", defaults: { archetype: archetype }
+    get "category/:category/none/l/top" => "list#category_none_top", as: "category_none_top#{archetype}", defaults: { archetype: archetype }
+    get "category/:parent_category/:category/l/top" => "list#parent_category_category_top", as: "parent_category_category_top#{archetype}", defaults: { archetype: archetype }
 
     TopTopic.periods.each do |period|
       get "top/#{period}" => "list#top_#{period}"
-      get "category/:category/l/top/#{period}" => "list#category_top_#{period}", as: "category_top_#{period}#{suffix}"
-      get "category/:category/none/l/top/#{period}" => "list#category_none_top_#{period}", as: "category_none_top_#{period}#{suffix}"
-      get "category/:parent_category/:category/l/top/#{period}" => "list#parent_category_category_top_#{period}", as: "parent_category_category_top_#{period}#{suffix}"
+      get "category/:category/l/top/#{period}" => "list#category_top_#{period}", as: "category_top_#{period}#{archetype}", defaults: { archetype: archetype }
+      get "category/:category/none/l/top/#{period}" => "list#category_none_top_#{period}", as: "category_none_top_#{period}#{archetype}", defaults: { archetype: archetype }
+      get "category/:parent_category/:category/l/top/#{period}" => "list#parent_category_category_top_#{period}", as: "parent_category_category_top_#{period}#{archetype}", defaults: { archetype: archetype }
     end
 
     Discourse.anonymous_filters.each do |filter|
-      get "#{filter}.rss" => "list##{filter}_feed", format: :rss
+      get "#{filter}.rss" => "list##{filter}_feed", format: :rss, defaults: { archetype: archetype }
     end
 
     Discourse.filters.each do |filter|
-      get "#{filter}" => "list##{filter}"
-      get "category/:category/l/#{filter}" => "list#category_#{filter}", as: "category_#{filter}#{suffix}"
-      get "category/:category/none/l/#{filter}" => "list#category_none_#{filter}", as: "category_none_#{filter}#{suffix}"
-      get "category/:parent_category/:category/l/#{filter}" => "list#parent_category_category_#{filter}", as: "parent_category_category_#{filter}#{suffix}"
+      get "#{filter}" => "list##{filter}", defaults: { archetype: archetype }
+      get "category/:category/l/#{filter}" => "list#category_#{filter}", as: "category_#{filter}#{archetype}", defaults: { archetype: archetype }
+      get "category/:category/none/l/#{filter}" => "list#category_none_#{filter}", as: "category_none_#{filter}#{archetype}", defaults: { archetype: archetype }
+      get "category/:parent_category/:category/l/#{filter}" => "list#parent_category_category_#{filter}", as: "parent_category_category_#{filter}#{archetype}", defaults: { archetype: archetype }
     end
   end
 
@@ -305,6 +305,7 @@ Discourse::Application.routes.draw do
     name = id if name.start_with?("translation missing:")
     scope "/#{name}" do
       categorizer(name)
+      get "" => "list#latest", defaults: { archetype: name }
     end
   end
 
-- 
1.8.3.4 (Apple Git-47)

